services/core-web/src/components/mine/ExplosivesPermit/ExplosivesPermit.js

```diff
@@ -212,54 +219,44 @@ export class ExplosivesPermit extends Component {
       });
   };
 
-  render() {
-    const title = this.props.isPermitTab
-      ? "Explosives Storage & Use Permit"
-      : "Explosives Storage & Use Permit Applications";
-    const data = this.props.isPermitTab
-      ? this.props.explosivesPermits.filter(
-          ({ application_status }) => application_status === "APP"
-        )
-      : this.props.explosivesPermits;
-    return (
-      <div>
-        <br />
-        <div className="inline-flex between">
-          <h4 className="uppercase">{title}</h4>
-          <AuthorizationWrapper
-            permission={
-              this.props.isPermitTab ? Permission.ADMIN : Permission.EDIT_EXPLOSIVES_PERMITS
-            }
-          >
-            <AddButton
-              onClick={(e) => this.handleOpenAddExplosivesPermitModal(e, this.props.isPermitTab)}
-            >
-              Add {title}
-            </AddButton>
-          </AuthorizationWrapper>
-        </div>
-        <br />
-        <MineExplosivesPermitTable
-          onExpand={this.onExpand}
-          expandedRowKeys={this.state.expandedRowKeys}
-          isLoaded
-          data={data}
-          isPermitTab={this.props.isPermitTab}
-          handleOpenExplosivesPermitDecisionModal={this.handleOpenExplosivesPermitDecisionModal}
-          handleOpenAddExplosivesPermitModal={this.handleOpenAddExplosivesPermitModal}
-          handleOpenViewMagazineModal={this.handleOpenViewMagazineModal}
-          explosivesPermitStatusOptionsHash={this.props.explosivesPermitStatusOptionsHash}
-          explosivesPermitDocumentTypeOptionsHash={
-            this.props.explosivesPermitDocumentTypeOptionsHash
-          }
-          handleOpenExplosivesPermitStatusModal={this.handleOpenExplosivesPermitStatusModal}
-          handleDeleteExplosivesPermit={this.handleDeleteExplosivesPermit}
-          handleOpenExplosivesPermitCloseModal={this.handleOpenExplosivesPermitCloseModal}
-        />
+  const title = isPermitTab
+    ? "Explosives Storage & Use Permit"
+    : "Explosives Storage & Use Permit Applications";
+  const data = isPermitTab
+    ? explosivesPermits.filter(({ application_status }) => application_status === "APP")
+    : explosivesPermits;
+
+  return (
+    <div>
+      <br />
+      <div className="inline-flex between">
+        <h4 className="uppercase">{title}</h4>
+        <AuthorizationWrapper permission={Permission.EDIT_EXPLOSIVES_PERMITS}>
```

Comment:
File refactored to functional component. The permission on the AuthorizationWrapper is the only real change.